admiral/pkg/clusters/serviceentry.go

```diff
@@ -208,10 +206,11 @@ func modifySidecarForLocalClusterCommunication(sidecarNamespace string, dependen
 	//copy and add our new local FQDN
 	newSidecar := copySidecar(sidecar)
 
-	egressHost := dependencyNamespace + "/" + localFqdn
-
-	if !util.Contains(newSidecar.Spec.Egress[0].Hosts, egressHost) {
-		newSidecar.Spec.Egress[0].Hosts = append(newSidecar.Spec.Egress[0].Hosts, egressHost)
+	for _, sidecarEgress := range sidecarEgressMap {
```

Comment:
I'm not against changing the SidecarEgressMap to be a map of maps, but you're not leveraging that here at all. You're still doing get(identity) then iterating through the values.

I guess my question then is why make the change to make the value another map, rather than a slice?

Reply:
`State before`
EgressMap had a mapping from a service identity -> List of Egresses
Every time we process a service, we add an entry into the List of Egress for all its dependents, meaning the list just keeps growing with duplicates
Now the modifySIdecar, the way it was being called also had a problem, you would call it per egress entry for the same sidecar N number of times, where N is size of the egress list

`State after this change`
EgressMap had a mapping from a service identity -> Map of Egresses by namespace
Every time we process a service, we add an entry and duplicates will be filtered
modifySidecar still has to iterate over all egress entries but only the ones not already present will be added and sidecar will be updated only once per source service